operators/src/source/gdal_source.rs

```diff
@@ -619,6 +638,67 @@ where
     Grid::new(tile_grid, buffer.data, no_data_value).map_err(Into::into)
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct GdalMetadataMapping {
+    source_key: RasterPropertiesKey,
+    target_key: RasterPropertiesKey,
+    target_type: RasterPropertiesEntryType,
+}
+
+fn properties_from_gdal<'a, I, M>(
+    properties: &mut RasterProperties,
```

Comment:
Maybe just consume it, idk

Reply:
please elaborate

Reply:
You could consume it and return a new instance if you modified it.

Reply:
but is that a cleaner way to do it?